go-selinux/label/label_selinux.go

```diff
@@ -118,7 +119,19 @@ func SocketLabel() (string, error) {
 // SetKeyLabel takes a process label and tells the kernel to assign the
 // label to the next kernel keyring that gets created
 func SetKeyLabel(processLabel string) error {
-	return selinux.SetKeyLabel(processLabel)
+	if processLabel == "" && selinux.GetEnabled() {
+		if _, err := KeyLabel(); err == io.EOF {
+			// If we got io.EOF err, this means it is the first time write "" to keycreate
+			// So we can ignore it
+			return nil
+		}
+	}
+	err := selinux.SetKeyLabel(processLabel)
```

Comment:
Why not put this check inside `selinux.SetKeyLabel`? Seems a bit odd to put it in this wrapper.

Reply:
Yes, it is.
If the user just use `"github.com/opencontainers/selinux/go-selinux"`, not `"github.com/opencontainers/selinux/go-selinux/label"`. 
I have update it.